pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -63,11 +63,17 @@ let
 
   # We can't use the existing fetchCrate function, since it uses a
   # recursive hash of the unpacked crate.
-  fetchCrate = pkg: fetchurl {
-    name = "crate-${pkg.name}-${pkg.version}.tar.gz";
-    url = "https://crates.io/api/v1/crates/${pkg.name}/${pkg.version}/download";
-    sha256 = pkg.checksum;
-  };
+  fetchCrate = pkg:
+    assert lib.assertMsg (pkg ? checksum) ''
+      package ${pkg.name} does not have a checksum.
+      Please note that the Cargo.lock format where checksums used to be listed under [metadata] is not supported.
+      If that is the case, running `cargo update` with a recent toolchain will automatically update the format along with the crate's depenendencies.
```

Comment:
Perhaps this can be formatted so that it fits nicely in 80 characters?

Reply:
done, I think? let me know if you meant something else